api/role.go

```diff
@@ -131,7 +154,10 @@ func (c CvpRestAPI) AddRole(role *SingleRole) error {
 			retErr = errors.Errorf("AddRole: %s", err)
 		}
 	}
-	return retErr
+	if retErr == nil {
+		return &returnedRole, nil
+	}
+	return nil, retErr
```

Comment:
you can do: 
_return &returnedRole, retErr_

Reply:
pointer to default object in not nil [`&returnedRole == nil`  will evaluate to false]

Let me change returnedRole to *SingleRole

Reply:
`return &returnedRole, retErr`does not work as I mentioned and changing  `var returnedRole *SingleRole` also does not work, because of Unmarshal function.